[15.0][MIG]sale_unreconciled

AaronHForgeFlow opened the pull request:
@ForgeFlow
Apart from the standard migration there is an improvement:

- [IMP] sale_unreconciled: add tolerance parameter

That way a company can block the locking when the amount unreconciled in the interim accounts is too high.

AaronHForgeFlow committed:
[14.0][ADD] sale_unreconciled

AaronHForgeFlow committed:
[IMP] sale_unreconciled: black, isort, prettier

AaronHForgeFlow committed:
[15.0][MIG] sale_unreconciled

JoanMForgeFlow reviewed (approved):
LG, code review seems to be OK!

AaronHForgeFlow committed:
[FIX] sale_unreconciled: do not create empty write-offs
[FIX] sale_unreconciled: ensure the accounting settigs are correct
[FIX] sale_unreconciled: assign sale order to the write-off created

[FIX] sale_unreconciled: search view

rafaelbn commented:
/ocabot migration sale_unreconciled

AaronHForgeFlow commented:
forever_alone.jpg

AaronHForgeFlow committed:
[15.0][FIX] sale_unreconciled: reconcile exit condition

AaronHForgeFlow committed:
[IMP] sale_unreconciled: allow companies to decide automatic reconcile or not

AaronHForgeFlow committed:
[FIX]sale_unreconciled: reconcile by currency

JuanyDForgeflow reviewed (approved):
LGTM!

AaronHForgeFlow committed:
[IMP] sale_unreconciled: add tolerance parameter

AaronHForgeFlow committed:
[FIX] sale_unreconciled: if journals have no residual but are full reconcile then assign a matching number
[IMP] sale_unreconciled: use the amount resiudal as condition for being unreconciled

AaronHForgeFlow committed:
[FIX] sale_unreconciled: multicompany for get unreconciled base domain

AaronHForgeFlow committed:
[FIX] sale_unreconciled: action to see unreconciled journal items has to consider the company of the SO

MiquelRForgeFlow commented on sale_unreconciled/models/sale_order.py:
```diff
@@ -0,0 +1,248 @@
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    unreconciled = fields.Boolean(
+        compute="_compute_unreconciled",
+        search="_search_unreconciled",
+        help="""Indicates that a sale Order has related Journal items not
+             reconciled.Note that if it is false it can be either that
+             everything is reconciled or that the related accounts do not
+             allow reconciliation""",
+    )
+    amount_unreconciled = fields.Float(
+        compute="_compute_unreconciled", compute_sudo=True
+    )
```
![Selection_718](https://user-images.githubusercontent.com/25005517/211852144-829b9100-1003-4998-b374-a44ec52564d9.png)

AaronHForgeFlow committed:
[IMP] sale_unreconciled: reconcile by groups

AaronHForgeFlow committed:
[FIX] sale_unreconcile: bad FW code

AaronHForgeFlow committed:
[FIX] sale_unreconciled: sudo in search ir.model

AaronHForgeFlow committed:
[IMP] sale_unreconciled: text

AaronHForgeFlow committed:
[IMP] sale_unreconciled: add hook for custom write-off date

AaronHForgeFlow committed:
[FIX] sale_unreconciled: unreconciled base domain to match standard

AaronHForgeFlow committed:
[IMP]sale_unreconciled: reconcile cancelled SO

AaronHForgeFlow committed:
fix sale_unrec

AaronHForgeFlow committed:
[FIX]sale_unreconciled: condition when no product

AaronHForgeFlow committed:
[IMP] sale_unreconciled: amount unreconciled in tree view

AaronHForgeFlow committed:
[FIX] sale_unreconciled: do the write-off on the latest date of the journals involved

[IMP] sale_unreconciled: choose residual field

AaronHForgeFlow committed:
[FIX] sale_unreconciled: do not show warning unreonciled message when reconcile criteria is not met
